src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectChangeEventArgs.cs

```diff
@@ -8,12 +8,12 @@ namespace Microsoft.CodeAnalysis.Razor.ProjectSystem;
 
 internal class ProjectChangeEventArgs : EventArgs
 {
-    public ProjectChangeEventArgs(IProjectSnapshot older, IProjectSnapshot newer, ProjectChangeKind kind)
+    public ProjectChangeEventArgs(ProjectSnapshot older, ProjectSnapshot newer, ProjectChangeKind kind)
```

Comment:
Please don't change `ProejctChangeEventsArgs` in this way. It definitely breaks the contract of `IProjectSnapshotManager` and is a shortcut that I'm uncomfortable with.

Instead, I recommend adding a `GetRelatedDocuments(...)` extension method on `IProjectSnapshot` that casts the receiver to `ProjectSnapshot` in order to call `ProjectSnapshot.GetRelatedDocuments(...)`. IMO, it's OK to throw if the receiver is not an instance of `ProjectSnapshot`.

Reply:
I considered going further and just making `ProjectSnapshotManager` deal exclusively with `ProjectSnapshot`, but managed to hold myself back. To me the event handler was okay because that seemed more like an implementation detail, but more than happy to entertain a different opinion. At least now that I know the fallout from changing the type was minimal, it gives confidence that the chance of hitting an exception is equally minimal, if not non-existent.

Reply:
> At least now that I know the fallout from changing the type was minimal, it gives confidence that the chance of hitting an exception is equally minimal, if not non-existent.

Agreed 100%.